### PR TITLE
Document fuzzer usage

### PR DESCRIPTION
**Pull request by nlohmann**
This PR adds documentation how to use LibFuzzer or afl++.

### CODE REVIEW

File: tests/fuzzing.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+for mutations.
+
+```shell
+TEST_DATA_VERSION=3.1.0
+wget https://github.com/nlohmann/json_test_data/archive/refs/tags/v$TEST_DATA_VERSION.zip
+unzip v$TEST_DATA_VERSION.zip
+rm v$TEST_DATA_VERSION.zip
+for FORMAT in json bjdata bson cbor msgpack ubjson
+do
+  rm -fr mkdir corpus_$FORMAT
```
Comment (falbrechtskirchinger) [inline tests/fuzzing.md:19]: Remove "mkdir".

File: tests/fuzzing.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+```
+
+The generated corpus can be used with both LibFuzzer and afl++.
+
+## LibFuzzer
+
+To use LibFuzzer, you need to pass `-fsanitize=fuzzer` as `FUZZER_ENGINE`:
+
+```shell
+make fuzzers FUZZER_ENGINE="-fsanitize=fuzzer"
```
Comment (falbrechtskirchinger) [inline tests/fuzzing.md:33]: If CXX isn't set in the current shell or doesn't point to `clang++` built with `libfuzzer`-enabled runtime, it has to be set here.

File: tests/fuzzing.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+The fuzzer should be able to run indefinitely without crashing. In case of a crash, the tested input is written to the
+directory `out`.
+
+## OSS-Fuzz
+
+The library is further fuzz-tested 24/7 by Google's [OSS-Fuzz project](https://github.com/google/oss-fuzz). It uses
+the same `fuzzers` target as above and also relies on the `FUZZER_ENGINE` variable. See the used
+[build script](https://github.com/google/oss-fuzz/blob/master/projects/json/build.sh) for more information.
+
+In case the build at OSS-Fuzz fails, and issue will be created automatically.
```
Comment (falbrechtskirchinger) [inline tests/fuzzing.md:79]: "... fails, an issue ..."

**Review by falbrechtskirchinger — CHANGES_REQUESTED** [review]
In addition to my review comments, you should also mention the working directory for the commands.

File: tests/fuzzing.md (hunk 3 above)
Comment (nlohmann) [reply tests/fuzzing.md]: Done: f92e6d0979190347796329a66b76fd9b16ad22a5

File: tests/fuzzing.md (hunk 2 above)
Comment (nlohmann) [reply tests/fuzzing.md:34]: Done: f92e6d0979190347796329a66b76fd9b16ad22a5

File: tests/fuzzing.md (hunk 1 above)
Comment (nlohmann) [reply tests/fuzzing.md]: Done: f92e6d0979190347796329a66b76fd9b16ad22a5

**Comment by nlohmann** [comment]
@falbrechtskirchinger Can I merge this? Or should we update the AFL wrapper with the one you used with AFL++?

**Comment by falbrechtskirchinger** [comment]
> @falbrechtskirchinger Can I merge this? Or should we update the AFL wrapper with the one you used with AFL++?

Let me give it another read, then you can merge it. We can update the AFL wrapper some other time.

I'd like to understand how the make targets are used by OSS-fuzz (haven't looked at `build.sh` which I've seen linked in comments) and maybe move the fuzz testing into the CMake code since the more advanced AFL++ use requires building two fuzzer binaries and then running the fuzzer requires a different command as well. All of that would be easier to do in CMake (we're also already downloading the test date which could simplify and automate corpus generation).

**Review by falbrechtskirchinger — APPROVED** [review]
Looks good to me.

**Comment by nlohmann** [comment]
Alright. I would like to avoid requiring OSS-Fuzz to use CMake. But the targets in the root `Makefile` can be removed/replaced.

**Pull request merged**